File: core/src/cpp/presents/net/UsernamePasswordCreds.cpp
#include "presents/stable.h"
#include "UsernamePasswordCreds.h"

using namespace presents::net;

DEFINE_STREAMABLE("com.threerings.presents.net.UsernamePasswordCreds", UsernamePasswordCreds);

void UsernamePasswordCreds::readObject (ObjectInputStream& in)
{
    presents::net::Credentials::readObject(in);
    username = boost::static_pointer_cast<util::Name>(in.readObject());
    password = in.readField< utf8 >();
}

void UsernamePasswordCreds::writeObject (ObjectOutputStream& out) const
{
    presents::net::Credentials::writeObject(out);
    out.writeObject(username);
    out.writeField(password);
}

File: core/src/main/java/com/threerings/presents/net/UsernamePasswordCreds.java
//
// Narya library - tools for developing networked games
// Copyright (C) 2002-2025 Three Rings Design, Inc., All Rights Reserved
// https://github.com/threerings/narya/blob/master/LICENSE

package com.threerings.presents.net;

import com.threerings.util.Name;

/**
 * Credentials that use a username and (hashed) password.
 */
public class UsernamePasswordCreds extends Credentials
{
    /**
     * Zero argument constructor used when unserializing an instance.
     */
    public UsernamePasswordCreds ()
    {
    }

    /**
     * Construct credentials with the supplied username and password.
     */
    public UsernamePasswordCreds (Name username, String password)
    {
        _username = username;
        _password = password;
    }

    public Name getUsername ()
    {
        return _username;
    }

    public String getPassword ()
    {
        return _password;
    }

    @Override
    public String getDatagramSecret ()
    {
        return _password;
    }

    @Override
    public String toString ()
    {
        StringBuilder buf = new StringBuilder("[");
        toString(buf);
        return buf.append("]").toString();
    }

    /**
     * An easily extensible method via which derived classes can add to {@link #toString}'s output.
     */
    protected void toString (StringBuilder buf)
    {
        buf.append("username=").append(_username);
        buf.append(", password=").append(_password);
    }

    protected Name _username;
    protected String _password;
}
